packages/@aws-cdk/cdk/lib/cloudformation/resource.ts

```diff
@@ -229,6 +232,29 @@ export class Resource extends Referenceable {
   }
 }
 
+export class TaggableResource extends Resource implements ITaggable {
+  /**
+   * TagManager to manage the propagation and assignment of tags
+   */
+  public readonly tags: TagManager;
+
+  constructor(parent: Construct, name: string, props: ResourceProps) {
+    super(parent, name, props);
+    // const initialTags = !!props.properties ? props.properties.tags || {} : {};
+    this.tags = new TagManager(this);
+  }
+
+  public toCloudFormation(): object {
+    const rawTags: Tag[] = resolve(this.properties.tags) || [];
+    const managedTags: Tag[] = resolve(this.tags) || [];
+    const rawTagKeys: string[] = rawTags.map(t => t.key);
+    // order is important L1 (raw) tags override managed tags
```

Comment:
Is this really the behavior we want? Perhaps we want to ensure that raw==managed in case there's a duplicate?

Reply:
I went back and forth here. 

The thought I had in my head was that if you created an L1 yourself you likely were 'overriding' something missing in our L2. However, I also coded the reverse, but did not consider the equality situation. I am handling the dedupe by tag key, so what do you mean in case there's duplication?